project/CONTRIBUTING.md

```diff
@@ -186,6 +186,24 @@ interface BarInterface
 }
 ```
 
+In some cases, you will have the possibility to warn the user that things will change,
+and recommend a new way of doing things. You can do so by triggering the dedicated kind of error, like this:
+
+```php
+<?php
+if (/* some condition showing the user is using the legacy way */) {
+    @trigger_error(
+        'The '.__METHOD.' method is deprecated since 42.x, to be removed in 43.0',
```

Comment:
`method is deprecated since version 3.1 and will be removed in 4.0.`

From Symfony code sample.

We can add also something like: `Use the Yaml::DUMP_EXCEPTION_ON_INVALID_TYPE flag instead.`.

-> https://github.com/symfony/symfony/blob/v3.1.0/src/Symfony/Component/Yaml/Dumper.php#L65